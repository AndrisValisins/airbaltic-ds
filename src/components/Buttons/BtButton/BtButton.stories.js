import BtButton from "./BtButton.vue";

export default {
  title: "Atoms/Buttons",
  component: BtButton,
  argTypes: {
    variant: {
      control: { type: "select" },
      options: ["primary", "secondary"],
    },
    onClick: { action: "clicked" },
  },
};

const Template = (args) => ({
  components: { BtButton },
  setup() {
    return { args };
  },
  template: `<bt-button v-bind="args">
    <template #iconBefore>{{ args.iconBefore }}</template>
    <template #default>{{ args.default }}</template>
    <template #iconAfter>{{ args.iconAfter }}</template>
  </bt-button>`,
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Primary.args = {
  default: "My primary button",
  variant: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  default: "Secondary",
  variant: "secondary",
};

export const WithIconBefore = Template.bind({});
WithIconBefore.args = {
  default: "Icon Before",
  variant: "primary",
  iconBefore: "A",
};

export const WithIconAfter = Template.bind({});
WithIconAfter.args = {
  default: "Icon After",
  variant: "primary",
  iconAfter: "A",
};

export const WithBothIcons = Template.bind({});
WithBothIcons.args = {
  default: "Icon Before",
  variant: "primary",
  iconAfter: "A",
  iconBefore: "A",
};

export const FullWidth = Template.bind({});
FullWidth.args = {
  default: "Full width",
  variant: "primary",
  full: true,
};
