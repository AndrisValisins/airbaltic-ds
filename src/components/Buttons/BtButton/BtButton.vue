<template>
  <button class="btn" :class="[variant, { 'full-width': full }]">
    <slot name="iconBefore" />
    <slot />
    <slot name="iconAfter" />
  </button>
</template>

<script lang="ts" setup>
defineProps({
  variant: {
    type: String,
    default: "primary",
    validator(value: string) {
      return ["primary", "secondary"].includes(value);
    },
  },
  full: {
    type: Boolean,
    default: false,
  },
});
</script>

<style lang="scss">
.btn {
  border: none;
  background: none;
  padding: 5px 10px;
  border-radius: 50px;
  color: white;
  cursor: pointer;

  &.primary {
    background: blue;
  }

  &.secondary {
    background: red;
  }

  &.full-width {
    min-width: 100%;
  }
}
</style>
