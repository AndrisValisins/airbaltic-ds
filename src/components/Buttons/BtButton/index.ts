export { default as BtButton } from "./BtButton.vue";
