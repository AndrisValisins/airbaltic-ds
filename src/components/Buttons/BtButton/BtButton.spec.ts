import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import BtButton from "./BtButton.vue";

describe("BaseButton", () => {
  it("Matches to snapshot", () => {
    const wrapper = mount(BtButton, {
      slots: {
        default: "Primary button",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders properly", () => {
    const wrapper = mount(BtButton, {
      slots: {
        default: "Primary button",
      },
    });
    expect(wrapper.text()).toContain("Primary button");
  });
});
