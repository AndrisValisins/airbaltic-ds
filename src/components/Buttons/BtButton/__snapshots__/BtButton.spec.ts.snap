// Vitest Snapshot v1

exports[`BaseButton > Matches to snapshot 1`] = `
<button
  class="btn primary"
>
  
  
  
  Primary button
  
  
  
</button>
`;
